core/eth/reader.go

```diff
@@ -804,64 +804,83 @@ func (t *Reader) GetAllVersionedBlobParams(ctx context.Context) (map[uint16]*cor
 	return res, nil
 }
 
-func (t *Reader) GetReservedPayments(ctx context.Context, accountIDs []gethcommon.Address) (map[gethcommon.Address]*core.ReservedPayment, error) {
+// TODO: later add quorumIds specifications
+func (t *Reader) GetReservedPayments(ctx context.Context, accountIDs []gethcommon.Address) (map[gethcommon.Address]map[core.QuorumID]*core.ReservedPayment, error) {
 	if t.bindings.PaymentVault == nil {
 		return nil, errors.New("payment vault not deployed")
 	}
-	reservationsMap := make(map[gethcommon.Address]*core.ReservedPayment)
-	reservations, err := t.bindings.PaymentVault.GetReservations(&bind.CallOpts{
-		Context: ctx,
-	}, accountIDs)
-	if err != nil {
-		return nil, err
-	}
 
-	// since reservations are returned in the same order as the accountIDs, we can directly map them
-	for i, reservation := range reservations {
-		res, err := ConvertToReservedPayment(reservation)
+	reservationsMap := make(map[gethcommon.Address]map[core.QuorumID]*core.ReservedPayment)
+
+	// In the new API, we need to query reservations per quorum and account
```

Comment:
For this code to work do we need the PaymentVault contract to be upgraded?

Reply:
yeah. the PaymentVault contract isn't merged yet. I have a separate PR on these changes but wanted to keep this PR contained to offchain metering